### PR TITLE
more epoch change

### PR DESCRIPTION
**Pull request by haroldcarr**
This completes (well, there are some TODO for `Broadcast`) the OBM specific changes to epoch change.

I'll handle the Broadcast stuff in a subsequent pull request.

### CODE REVIEW

File: LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-1.agda [hunk 1]
```diff
@@ -42,10 +96,17 @@ e_RoundState_processLocalTimeoutM e r =
 
 ------------------------------------------------------------------------------
 
-postulate -- TODO-1: e_EpochManager_doECP_waitForRlec
- e_EpochManager_doECP_waitForRlec
-  : EpochManager → EpochChangeProof
-  → Either ErrLog Bool
+e_EpochManager_doECP_waitForRlec : EpochManager → EpochChangeProof → Either ErrLog Bool
```
Comment (mark-moir) [inline LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-1.agda:99]: Eventually, before proving anything about this function, we'll want it in `EitherD`.  I'm OK with leaving that until then, but maybe add a TODO-1?

File: LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-1.agda [hunk 2]
```diff
@@ -75,17 +136,47 @@ e_EpochManager_startNewEpoch-D em = fromEither ∘ e_EpochManager_startNewEpoch
 
 ------------------------------------------------------------------------------
 
-postulate -- TODO-1: e_EpochManager_checkEpc
- e_EpochManager_checkEpc
-  : EpochManager → EpochChangeProof
-  → Either ErrLog Unit
+e_EpochManager_checkEpc : EpochManager → EpochChangeProof → Either ErrLog Unit
```
Comment (mark-moir) [inline LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-1.agda:139]: Eventually, before proving anything about this function, we'll want it in `EitherD`.  I'm OK with leaving that until then, but maybe add a TODO-1?

File: LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-1.agda [hunk 3]
```diff
@@ -75,17 +136,47 @@ e_EpochManager_startNewEpoch-D em = fromEither ∘ e_EpochManager_startNewEpoch
...
+  here' t = "e_EpochManager_checkEpc" ∷ t
+
 
 ------------------------------------------------------------------------------
 
-postulate -- TODO-1: e_EpochManager_processMessage_ISyncInfo
- e_EpochManager_processMessage_ISyncInfo
-  : EpochManager → SyncInfo
-  → Either ErrLog Unit
+e_EpochManager_processMessage_ISyncInfo : EpochManager → SyncInfo → Either ErrLog Unit
```
Comment (mark-moir) [inline LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-1.agda:172]: Eventually, before proving anything about this function, we'll want it in `EitherD`.  I'm OK with leaving that until then, but maybe add a TODO-1?

File: LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-2.agda [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+-- This is a separate file to avoid a cyclic dependency.
+
+------------------------------------------------------------------------------
+
+e_DiemDB_getEpochEndingLedgerInfosImpl_limit : Epoch → Epoch → Epoch → (Epoch × Bool)
+e_DiemDB_getEpochEndingLedgerInfosImpl_limit startEpoch endEpoch limit =
+  if not ECP-LBFT-OBM-Diff-0.enabled
+  then
+    if-dec endEpoch ∸ startEpoch >? limit
```
Comment (mark-moir) [inline LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-2.agda:30]: If we rewrite this as `if-dec endEpoch >? startEpoch + limit` (and similarly in Haskell), we avoid the monus and reasoning about it, FWIW.

**Review by mark-moir — APPROVED** [review]
Just a couple of small comments, otherwise LGTM.  When you've addressed them to your satisfaction, APPROVED.  Thanks @haroldcarr.

File: LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-1.agda (hunk 1 above)
Comment (haroldcarr) [reply LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-1.agda:99]: OK

File: LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-1.agda (hunk 2 above)
Comment (haroldcarr) [reply LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-1.agda:139]: OK

File: LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-2.agda (hunk 4 above)
Comment (haroldcarr) [reply LibraBFT/Impl/OBM/ECP-LBFT-OBM-Diff/ECP-LBFT-OBM-Diff-2.agda:30]: DONE.  In Haskell too.

**Pull request merged**